test/jeison-test.el

```diff
@@ -218,4 +234,148 @@
+                 (and (not (null client-data))
+                      (null (gethash "non-existent" client-table)))))))))
+
+(ert-deftest jeison:check-hash-table-of-hash-table-of-type ()
+  (jeison-defclass jeison:version-thing ()
+    ((major :initarg :major :initform 0 :type integer)
+     (minor :initarg :minor :initform 0 :type integer)))
+  (jeison-defclass jeison:jeison-thing nil
+    ((jsonFile :initarg :jsonFile :initform "invalid")
+     (kind :initarg kind :initform "nothing")
+     (version :initarg :version :type jeison:version-thing)))
+  (jeison-defclass jeison:jeison-class nil
+    ((reply :initarg :reply :type (jeison-hash-table-of symbol (jeison-hash-table-of symbol jeison:jeison-thing)))))
+  (let ((answer (jeison-read jeison:jeison-class "{
+        \"reply\" :
+        {
+            \"client-1aa66f801e65edda\" :
+            {
+                \"codemodel-v2\" :
+                {
+                    \"jsonFile\" : \"codemodel-v2-bd9606760e25788500f7.json\",
+                    \"kind\" : \"codemodel\",
+                    \"version\" :
+                    {
+                        \"major\" : 2,
+                        \"minor\" : 4
+                    }
+                }
+            },
+            \"client-1aa66f801e65beef\" :
+            {
+                \"codemodel-v2\" :
+                {
+                    \"jsonFile\" : \"codemodel-v2-bd9606760e292887f0f7.json\",
+                    \"kind\" : \"codemodel\",
+                    \"version\" :
+                    {
+                        \"major\" : 2,
+                        \"minor\" : 4
+                    }
+                }
+            }
+        }
+}")))
+    (should (with-slots ((client-table reply)) answer
+              (let ((client-1 (gethash 'client-1aa66f801e65edda client-table))
+                    (client-2 (gethash 'client-1aa66f801e65beef client-table)))
+                (and (not (null client-1))
+                     (not (null (gethash 'codemodel-v2 client-1)))
+                     (not (null client-2))
+                     (null (gethash 'non-existent client-table))
+
+                     ;; All of these tests succeed: puthash isn't picky about
+                     ;; what it stashes. The caveat: the hash table uses 'eq'
+                     ;; for comparisons.
+                     (puthash "client-123456" "client-123456 data" client-table)
+                     (puthash "client-7890ab" [1 2 3 4 5] client-table)
+                     (puthash [1 3 5] "vector data" client-table)
+                     ;; These 'gethash'-es fail because strings don't compare
+                     ;; with 'eq'.
+                     (null (gethash "client-123456" client-table))
+                     (null (gethash "client-7890ab" client-table))
+                     ;; And neither do vectors.
+                     (null (gethash [1 3 5] client-table))))))))
```

Comment:
Now this is pure trolling. Try listen to other people instead of trying to mentor them (I'm mentoring you here).
Jeison reading functions DO TYPE CHECKING, you introduced NEW TYPES, test that the type checking for new types is WORKING when we read from JSON.